test/e2e/framework/kcp.go

```diff
@@ -326,11 +326,20 @@ func (c *kcpServer) waitForEndpoint(client *rest.RESTClient, endpoint string) {
 			lastMsg = fmt.Sprintf("error contacting %s: %v", endpoint, err)
 		}
 	}, 100*time.Millisecond)
-	if !succeeded {
+	if !succeeded && !isDone(c.ctx) {
 		c.t.Errorf(lastMsg)
 	}
 }
 
+func isDone(ctx context.Context) bool {
```

Comment:
Nit: as per the Go language spec we should do `if ctx.Error() != nil`

Reply:
fixed